Rubberduck.VBEEditor/WindowsApi/FocusSource.cs

```diff
@@ -8,11 +8,18 @@ internal abstract class FocusSource : SubclassingWindow, IWindowEventProvider
         protected FocusSource(IntPtr hwnd) : base(hwnd, hwnd) { }
```

Comment:
I don't like that I'm making the abstract class a misnommer now.

Reply:
There is also the issue that you are building a class hierarchy where there shouldn't be a hierarchy. From Win32 POV, there can be only one window procedure, and since we are using this to create services for different window messages (e.g. listening to certain messages to interpret as focus-related events, other messages as keyboard-related events), we should be able to compose, not inherit the window procedure, I think.

Reply:
FWIW the class hierarchy was there to begin with, I just made it respond to more WMs..